include/aws/cryptosdk/multi_keyring.h

```diff
@@ -22,6 +22,13 @@
 extern "C" {
 #endif
 
+struct multi_keyring {
+    struct aws_cryptosdk_keyring base;
+    struct aws_allocator *alloc;
+    struct aws_cryptosdk_keyring *generator;
+    struct aws_array_list children;  // list of (struct aws_cryptosdk_keyring *)
+};
+
```

Comment:
@tawdry-audrey could you confirm whether we can move `struct multi_keyring`declaration to the `.h` file?

Reply:
Its possible that this is meant to be an opaque type.  In that case, we should create a new private/multi_keyring.h

Reply:
I moved to the private folder.